src/capi/azext_capi/custom.py

```diff
@@ -706,17 +707,47 @@ def get_kubeconfig(capi_name):
     return f"Wrote kubeconfig file to {filename} "
 
 
-def delete_workload_cluster(cmd, capi_name, yes=False):
+def delete_workload_cluster(cmd, capi_name, resource_group_name=None, yes=False):
     exit_if_no_management_cluster()
     msg = f'Do you want to delete this Kubernetes cluster "{capi_name}"?'
+    command = ["kubectl", "delete", "cluster", capi_name]
+    is_self_managed = is_self_managed_cluster(capi_name)
+    if is_self_managed:
+        while not resource_group_name:
+            resource_group_name = prompt_method(msg=f'Please enter resource group name of "{capi_name}" cluster: ')
```

Comment:
do we need that info from the user or can we guess it from the AzureCluster object?